### ISSUE TITLE
High-Level Introduction Structure

### ISSUE THREAD

**Issue by samm82** [samm82/TestGen-Thesis#126]
[The resource linked to](https://www.monash.edu/student-academic-success/excel-at-writing/how-to-write/thesis-chapter/introduction-and-conclusion-thesis-chapters) in #111 lists the following features of a thesis intro:

1. Introduce the topic
2. Define key terms and concepts
3. Give background and context for the topic (this may include a brief literature review)
4. Review and evaluate the current state of knowledge in the topic (this may include a brief literature review)
5. Identify any gaps, shortcomings and problems in the research to date
6. Introduce your research question(s) or hypothesis
7. Briefly describe your methodology and/or theoretical approach
8. Explain the aim of your research and what contribution it will make to the topic
9. Give an overview of the chapter outline of the thesis

I am a little unsure as to what structural changes I should be making based on this list. Was the issue that I was too unclear? The following list is either rationale for how it's already included or questions on how I should be including it, following the same numbering scheme:
1. This is the introductory bit talking about software testing itself. I agree that more justification should be added to my claims (this was also a critique of my paper which I have yet to address; probably should have already been done 😬) and this overview of software testing will likely be expanded to its own paragraph (#127)
2. This is currently done in the a subsection of Methodology, since some of it came up organically during the research phase itself. Should I move it here, or at least a high-level overview?
3. The background and context is composed of:
   1. Arguments for the possibility of miscommunication arising from unclear terminology
   2. Our Drasil-based motivation; you say I should "separate [it] out as 'motivation'". Does that mean it should be its own subsection, just a new paragraph, or at the very least more "atomic" (i.e., clear that this just background knowledge and not a justification for the research itself)?
4. I try to accomplish this through three approaches:
   1. I overview existing taxonomies along with why they aren't sufficient for us. This is why "those sources are singled out"; is that what your confusion was about? I could make that more explicit (a starting point could be something like "the following taxonomies were identified, but were too narrow in scope for our purposes").
   3. I provide trivial examples of the fallibility of existing literature. You mentioned these are "dumb mistakes---bad leading example[s]!", but I thought @smiths had mentioned previously (#116) that these "dumb mistakes" are good examples to include up front! Did you mean that these should be moved or *re*moved?
   4. I provide some self-contained examples of how these types of mistakes permeate the field of software testing.
5. Done alongside feature 4, above.
6. Done in bulleted list.
7. Done in paragraph after feature 6, above; I will expand on this alongside the work I've been doing based on #112. 
8. Done in Declaration of Academic Achievement (#102).
9. Done in final paragraph.

**Comment by JacquesCarette** [comment]
You don't have to follow the resource "on the nose", but use it as a guide to better structure and think through your introduction.

In some sense, your 'topic' is actually *incoherent definitions* in the sub-domain of SE that is 'software testing'. So the first paragraph might well cover SE -> testing as part of SE -> but what is testing.

Again, it might make sense to think of the key terms for your actual thesis to be 'definition', 'taxonomy', 'coherent', 'ontology', even 'etymology'!

The background & context can include both the original motivation (generating tests) and the eventual project: finding the many places where the literature gives flawed information on the topic of testing. The "lit. review" here would include listing the standards and the various 'meta' documents that attempt to collate definitions, terminology, etc.

And so on.

What's wrong with the current introduction is that it won't make sense to anyone who doesn't already know what your thesis is about. That is your target readership. You need to convince your reader that what you've done is interesting.

Basically: you don't seem to have "designed" your introduction, but just written it. I don't feel like you had a series of 'points' that you needed to make (to the reader) but rather, you were *telling* the reader what you'd done.

**Comment by samm82** [comment]
I took another shot at restructuring this intro, starting more generally from the field of software engineering instead of focusing on testing right away. The relevant content is the first three paragraphs in the attached PDF below. I also remembered that I took some notes on "Roadblocks to Testing"; would these be helpful to include (and expand on) in the intro, or would that be getting too specific too quickly? @JacquesCarette 

[Sam-restructured-intro.pdf](https://github.com/user-attachments/files/18216765/Sam-restructured-intro.pdf)

![image](https://github.com/user-attachments/assets/4d734c1a-13ac-45e7-8671-ee55f6966dfa)

**Comment by JacquesCarette** [comment]
Wow, this is a massive improvement! I really like this new introduction (I read the whole thing that is in 'thesis.pdf' in the current repo). I really have no comments on it - which is rare.

I don't think there is a need to add these "roadblocks to testing" to the introduction. I think it does a good job as is.
